packages/worker/src/utilities/email.ts

```diff
@@ -158,10 +143,21 @@ export async function sendEmail(
   }
   const transport = createSMTPTransport(config)
   // if there is a link code needed this will retrieve it
-  const code = await getLinkCode(purpose, email, opts.user, opts?.info)
+  let code: string | null = null
+  switch (purpose) {
+    case EmailTemplatePurpose.PASSWORD_RECOVERY:
+      if (!opts.user || !opts.user._id) {
+        throw new HTTPError("User must be provided for password recovery.", 400)
+      }
+      code = await cache.passwordReset.createCode(opts.user._id, opts.info)
+      break
+    case EmailTemplatePurpose.INVITATION:
+      code = await cache.invite.createCode(email, opts.info)
+      break
+  }
```

Comment:
This got moved from `getLinkCode` and I added the checks to `opts.user` because the newly corrected typing was showing the user of `opts.user` to be incorrect.

Reply:
Out of interest, what was the reason for lifting it out of `getLinkCode`?

Reply:
`getLinkCode` was only used here and had as many arguments as it did significant lines of code. It's probably a matter of taste more than anything, but it felt like unnecessary indirection to me.

Reply:
Was the `getLinkCode` function removed in that case? If it is no longer used we should get rid of it (although it maybe used in the account portal potentially)

Reply:
It was, yes 🙂 It was only in this file and not exported.